baseapp/baseapp_test.go

```diff
@@ -39,39 +39,32 @@ var (
 
 type paramStore struct {
 	db *dbm.MemDB
+	t  *testing.T
 }
 
 func (ps *paramStore) Set(_ sdk.Context, key []byte, value interface{}) {
 	bz, err := json.Marshal(value)
-	if err != nil {
-		panic(err)
-	}
-
-	ps.db.Set(key, bz)
+	require.NoError(ps.t, err)
+	err = ps.db.Set(key, bz)
```

Comment:
This is out of scope of the functionality undergoing testing.

Reply:
Why? This task is to catch unhandled errors.

Reply:
Testing database returned errors is out of scope here. This is meant to test `baseapp`, and not `MemDB`

Reply:
I see only benefits for detecting errors rather than ignoring them. Especially in integration tests.

Reply:
These are unit tests

Reply:
Is the errcheck linter flagging this? I also don't really see the problem with this method panic'ing in a unit test. I agree with checking errors in general, but sometimes if you are implementing an interface and the method doesn't return `error`, panic is the only option no? So I'm not sure what this buys us...

Reply:
> These are unit tests

@alessio - what's your definition for unit tests? According to mine (same as [here](https://www.techopedia.com/definition/9847/unit-test)) there is no unit test in `baseapp_test.go`.

Reply:
@aaronc  - yes, it's flagging it. Here is the list: https://pastebin.com/AY6ATduY. 

Handling errors with `*testing.T` is better than using `panic` - much better messages, you can use all the goodies from `require`...

> but sometimes if you are implementing an interface and the method doesn't return error, panic is the only option no?

This is usually a signal that the interface is dangerous or has a tech debt  and it's probably better to update the interface. Probably all IO methods should return error. Notice that some interfaces (eg related to

Reply:
Okay that sounds reasonable even if it's a little awkward

Reply:
Why awkward? In this particular example we wrap the storage anyway for tests, and the DB can return error, so we can check that with a testing context. For me it's an easy / elegant way how to handle this errors in tests (rather than obfuscating the _Test*_ functions).

Reply:
I agree with Aaron here, it's not entirely unreasonable - yet this seems awkward and smells like an overkill.

> Probably all IO methods should return error

This test cases use an in-memory database - again, a `panic()` would just do it and all the goodies that come with `require` bring very little benefit.

Reply:
@alessio  Why do you prefer to use `panic` rather than `require`? In both cases the test will stop, and with `require` we will have a better, more friendly message. Now (in master) this is left undetected.

Reply:
OK, if it's in-memory then it _shouldn't_ return error. It's still better to check it. Also, if someone, for whatever reason, will change the DB, then it will cause bigger problems.

Reply:
I can't foresee a reason why we would want to use something different than an in-memory database for unit tests - can you?

However, fine - let's move on.

Reply:
When doing this updates, I don't spend much time thinking what dependencies  we are using.  I think the rule to "check errors" is simple and safe. Not checking errors may lead to problems. Let's stick to the "simple and safe".

Reply:
Let's move.